lib/gs1/ai.rb

```diff
@@ -0,0 +1,294 @@
+    GLN_EXTENSION_COMPONENT = '254'.freeze
+
+    # Global Coupon Number (GCN)
+    GCN = '255'.freeze
+
+    #
+    # GS1 Application Identifiers starting with digit 3
+
+    # Count of items (variable measure trade item)
+    VAR_COUNT = '30'.freeze
+
+    #
+    # GS1 Application Identifiers starting with digit 4
+
+    # Customer's purchase order number
+    ORDER_NUMBER = '400'.freeze
+
+    # Global Identification Number for Consignment (GINC)
+    GINC = '401'.freeze
+
+    # Global Shipment Identification Number (GSIN)
+    GSIN = '402'.freeze
+
+    # Routing code
+    ROUTE = '403'.freeze
+
+    # Ship to - Deliver to Global Location Number
+    SHIP_TO_LOC = '410'.freeze
+
+    # Bill to - Invoice to Global Location Number
+    BILL_TO = '411'.freeze
+
+    # Purchased from Global Location Number
+    PURCHASE_FROM = '412'.freeze
+
+    # Ship for - Deliver for - Forward to Global Location Number
+    SHIP_FOR_LOC = '413'.freeze
+
+    # Identification of a physical location - Global Location Number
+    LOC_NO = '414'.freeze
+
+    # Global Location Number of the invoicing party
+    PAY_TO = '415'.freeze
+
+    # GLN of the production or service location
+    PROD_SERV_LOC = '416'.freeze
+
+    # Ship to - Deliver to postal code within a single postal authority
+    SHIP_TO_POST = '420'.freeze
+
+    # Ship to - Deliver to postal code with ISO country code
+    SHIP_TO_POST_ISO = '421'.freeze
+
+    # Country of origin of a trade item
+    ORIGIN = '422'.freeze
+
+    # Country of initial processing
+    COUNTRY_INITIAL_PROCESS = '423'.freeze
+
+    # Country of processing
+    COUNTRY_PROCESS = '424'.freeze
+
+    # Country of disassembly
+    COUNTRY_DISASSEMBLY = '425'.freeze
+
+    # Country covering full process chain
+    COUNTRY_FULL_PROCESS = '426'.freeze
+
+    # Country subdivision Of origin
+    ORIGIN_SUBDIVISION = '427'.freeze
+
+    #
+    # GS1 Application Identifiers starting with digit 7
+
+    # NATO Stock Number (NSN)
+    NSN = '7001'.freeze
+
+    # UN/ECE meat carcasses and cuts classification
+    MEAT_CUT = '7002'.freeze
+
+    # Expiration date and time
+    EXPIRY_TIME = '7003'.freeze
+
+    # Active potency
+    ACTIVE_POTENCY = '7004'.freeze
+
+    # Catch area
+    CATCH_AREA = '7005'.freeze
+
+    # First freeze date
+    FIRST_FREEZE_DATE = '7006'.freeze
+
+    # Harvest date
+    HARVEST_DATE = '7007'.freeze
+
+    # Species for fishery purposes
+    AQUATIC_SPECIES = '7008'.freeze
+
+    # Fishing gear type
+    FISHING_GEAR_TYPE = '7009'.freeze
+
+    # Production method
+    PROD_METHOD = '7010'.freeze
+
+    # Refurbishment lot ID
+    REFURB_LOT = '7020'.freeze
+
+    # Functional status
+    FUNC_STAT = '7021'.freeze
+
+    # Revision status N4+X..20 (FNC1)
+    REV_STAT = '7022'.freeze
+
+    # Global Individual Asset Identifier (GIAI) of an assembly
+    GIAI_ASSEMBLY = '7023'.freeze
+
+    # Number of processor with ISO Country Code N4+N3+X..27 (FNC1)
+    PROCESSOR = '703s'.freeze
```

Comment:
All other are pure numbers, is this `s` intended?

Reply:
Removed!